langchain/llms/ai21.py

```diff
@@ -0,0 +1,120 @@
+
+    temperature: float = 0.7
+    """What sampling temperature to use."""
+
+    maxTokens: int = 256
+    """The maximum number of tokens to generate in the completion."""
+
+    minTokens: int = 0
+    """The minimum number of tokens to generate in the completion."""
+
+    topP: float = 1.0
+    """Total probability mass of tokens to consider at each step."""
+
+    presencePenalty: AI21PenaltyData = AI21PenaltyData()
+    """Penalizes repeated tokens."""
+
+    countPenalty: AI21PenaltyData = AI21PenaltyData()
+    """Penalizes repeated tokens according to count."""
+
+    frequencyPenalty: AI21PenaltyData = AI21PenaltyData()
+    """Penalizes repeated tokens according to frequency."""
+
+    numResults: int = 1
+    """How many completions to generate for each prompt."""
+
+    logitBias: Dict[str, float] = None
+    """Adjust the probability of specific tokens being generated."""
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key exists in environment."""
+        if "AI21_API_KEY" not in os.environ:
+            raise ValueError(
+                "Did not find AI21 API key, please add an environment variable"
+                " `AI21_API_KEY` which contains it."
+            )
+        return values
+
+    @property
+    def _default_params(self) -> Mapping[str, Any]:
+        """Get the default parameters for calling AI21 API."""
+        return {
+            "temperature": self.temperature,
+            "maxTokens": self.maxTokens,
+            "minTokens": self.minTokens,
+            "topP": self.topP,
+            "presencePenalty": self.presencePenalty.dict(),
+            "countPenalty": self.countPenalty.dict(),
+            "frequencyPenalty": self.frequencyPenalty.dict(),
+            "numResults": self.numResults,
+            "logitBias": self.logitBias,
+        }
+
+    def __call__(self, prompt: str, stop: Optional[List[str]] = None) -> str:
+        """Call out to AI21's complete endpoint.
+
+        Args:
+            prompt: The prompt to pass into the model.
+            stop: Optional list of stop words to use when generating.
+
+        Returns:
+            The string generated by the model.
+
+        Example:
+            .. code-block:: python
+
+                response = ai21("Tell me a joke.")
+        """
+        if stop is None:
+            stop = []
+        response = requests.post(
+            url=f"https://api.ai21.com/studio/v1/{self.model}/complete",
+            headers={"Authorization": f"Bearer {os.environ['AI21_API_KEY']}"},
+            json={
+                "prompt": prompt,
+                "stopSequences": stop,
+                **self._default_params,
+            }
+        )
+        assert response.status_code == 200, f"AI21 /complete call failed with status code {response.status_code}"
```

Comment:
can we do 
```
if response.status_code != 200:
     raise ValueError(...)
```
instead?

per google style https://google.github.io/styleguide/pyguide.html am only trying to use asserts for internal correctness, and since this is an external api call would like to use value error here

Reply:
Sure no problem. I'm on it

Reply:
Done.